crates/curp/src/server/storage/wal/segment.rs

```diff
@@ -156,23 +127,62 @@ impl WALSegment {
     /// Seal the current segment
     ///
     /// After the seal, the log index in this segment should be less than `next_index`
-    pub(super) async fn seal<C: Serialize>(&mut self, next_index: LogIndex) -> io::Result<()> {
-        let mut framed = Framed::new(self, WAL::<C>::new());
-        framed.send(vec![DataFrame::SealIndex(next_index)]).await?;
-        framed.flush().await?;
-        framed.get_mut().sync_all().await?;
-        framed.get_mut().update_seal_index(next_index);
+    pub(super) fn seal<C: Serialize>(&mut self, next_index: LogIndex) -> io::Result<()> {
+        self.write_sync(vec![DataFrame::SealIndex(next_index)], WAL::<C>::new())?;
+        self.update_seal_index(next_index);
         Ok(())
     }
 
-    /// Syncs the file of this segment
-    pub(super) async fn sync_all(&mut self) -> io::Result<()> {
-        self.file.sync_all().await?;
-        self.io_state.fsynced();
+    /// Writes an item to the segment
+    pub(super) fn write_sync<U, Item>(&mut self, item: Item, mut encoder: U) -> io::Result<()>
+    where
+        U: Encoder<Item, Error = io::Error>,
+    {
+        let encoded_bytes = encoder.encode(item)?;
+        self.file.write_all(&encoded_bytes)?;
+        self.update_size(encoded_bytes.len().numeric_cast());
```

Comment:
Why not syncing file first, then updating size, in case any error happens?

Reply:
fixed